orange/src/main/java/com/hartwig/hmftools/orange/algo/util/GermlineConversion.java

```diff
@@ -147,6 +161,96 @@ private static List<PurpleGainLoss> toSomaticGainsLosses(@Nullable List<PurpleGa
         return reportableGermlineGainsLosses != null ? reportableGermlineGainsLosses : Lists.newArrayList();
     }
 
+    @NotNull
+    private static List<PurpleGeneCopyNumber> getAdditionalSuspectGeneCopyNumberWithLOH(@NotNull PurpleRecord purple)
+    {
+        List<PurpleHeterozygousDeletion> reportableGermlineHeterozygousDeletions = purple.reportableGermlineHeterozygousDeletions();
+        if(reportableGermlineHeterozygousDeletions == null)
+        {
+            return Lists.newArrayList();
+        }
+
+        List<String> relevantGenes = reportableGermlineHeterozygousDeletions.stream()
+                .map(PurpleHeterozygousDeletion::gene)
+                .distinct()
+                .collect(Collectors.toList());
+
+        List<PurpleGeneCopyNumber> additionalSuspectGeneCopyNumberWithLOH = Lists.newArrayList();
+        for(String gene : relevantGenes)
+        {
+            PurpleGeneCopyNumber candidate = findMatchingGeneCopyNumber(gene, purple.allSomaticGeneCopyNumbers());
+
+            if(candidate != null
+                    && shouldAddSuspectGeneCopyNumberWithLOH(candidate, purple.suspectGeneCopyNumbersWithLOH(), purple.reportableGermlineFullLosses()))
+            {
+                PurpleGeneCopyNumber adjustedGeneCopyNumber = correctForGermlineImpact(candidate, reportableGermlineHeterozygousDeletions);
+                additionalSuspectGeneCopyNumberWithLOH.add(adjustedGeneCopyNumber);
+            }
+        }
+        return additionalSuspectGeneCopyNumberWithLOH;
+    }
+
+    private static boolean shouldAddSuspectGeneCopyNumberWithLOH(@NotNull PurpleGeneCopyNumber candidateSuspectGeneCopyNumber,
+            @NotNull List<PurpleGeneCopyNumber> suspectGeneCopyNumbersWithLOH, @Nullable List<PurpleGainLoss> reportableGermlineFullLosses)
+    {
+        String gene = candidateSuspectGeneCopyNumber.geneName();
+
+        boolean alreadySuspectGeneCopyNumberWithLOH = findMatchingGeneCopyNumber(gene, suspectGeneCopyNumbersWithLOH) != null;
+        boolean hasReportableGermlineFullLoss =
+                reportableGermlineFullLosses != null && reportableGermlineFullLosses.stream().anyMatch(l -> l.gene().equals(gene));
+        boolean notFullyLostInTumor = notFullyLostInTumor(candidateSuspectGeneCopyNumber);
+
+        return !alreadySuspectGeneCopyNumberWithLOH && !hasReportableGermlineFullLoss && notFullyLostInTumor;
+    }
+
+    @Nullable
+    private static PurpleGeneCopyNumber findMatchingGeneCopyNumber(@NotNull String gene,
+            @NotNull List<PurpleGeneCopyNumber> geneCopyNumbers)
+    {
+        for(PurpleGeneCopyNumber geneCopyNumber : geneCopyNumbers)
+        {
+            if(geneCopyNumber.geneName().equals(gene))
+            {
+                return geneCopyNumber;
+            }
+        }
+        return null;
+    }
+
+    private static boolean notFullyLostInTumor(@NotNull PurpleGeneCopyNumber geneCopyNumber)
```

Comment:
Similar to before, would not negate the function itself. Would refactor to `!fullyLostInTumor`